### PR TITLE
fix(core): format_units overflow

### PR DESCRIPTION
**Pull request by DaniPopes**
## Motivation

U256::from(u128...);
U256::exp10(x) with x > 77;
also lots of unnecessary calculations and casts

## PR Checklist

-   [ ] Added Tests
-   [ ] Added Documentation
-   [ ] Updated the changelog

### CODE REVIEW

**Review by mattsse — COMMENTED** [review]
nice catch, can we get a smol test for these edge cases?

File: ethers-core/src/utils/mod.rs [hunk 1]
```diff
@@ -158,29 +158,32 @@ where
...
 {
-    let units = units.try_into()?;
-    match amount.into() {
+    let units: usize = units.try_into()?.into();
+    let amount = amount.into();
+
+    // 78 overflows U256, 77 overflows I256
+    match amount {
+        // 2**256 ~= 1.16e77
+        ParseUnits::U256(_) if units >= 78 => return Err(ConversionError::ParseOverflow),
```
Comment (0xMelkor) [inline ethers-core/src/utils/mod.rs:167]: Despite the comments `77` and `78` sound like magic numbers. Should we assign constants with meaningful names?

Comment (gakonst) [reply ethers-core/src/utils/mod.rs:167]: +1

**Review by gakonst — APPROVED** [review]
Love it - let's move the magic nums in consts and merge
